src/Microsoft.DotNet.Interactive/Parsing/DirectiveParser.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Parsing;
+
+namespace Microsoft.DotNet.Interactive.Parsing
+{
+    internal class DirectiveParser
+    {
+        private readonly Parser _commandLineParser;
+        public DirectiveParser(RootCommand rootCommand)
+        {
+            var commandLineBuilder =
+                    new CommandLineBuilder(rootCommand)
+                        .ParseResponseFileAs(ResponseFileHandling.Disabled)
+                        .UseTypoCorrections()
+                        .UseHelpBuilder(bc => new DirectiveHelpBuilder(rootCommand.Name))
+                        .UseHelp()
+                        .UseMiddleware(
+                            context =>
+                            {
+                                context.BindingContext
+                                       .AddService(
+                                           typeof(KernelInvocationContext),
+                                           _ => KernelInvocationContext.Current);
+                            });
+            commandLineBuilder.EnableDirectives = false;
+            _commandLineParser = commandLineBuilder.Build();
+        }
+
+        public ParseResult Parse(string input)
+        {
+            if (FindComment(input) is { } realPos)
+                input = input.Substring(0, realPos);
+            return _commandLineParser.Parse(input);
+        }
+
+        private static int? FindComment(string input)
```

Comment:
Oh, and unlike the rest if the code, this part... well, it might be implemented better than I did, so any recommendations are welcomed.